src/main.rs

```diff
@@ -47,10 +42,25 @@ impl State {
 }
 
 #[embassy_executor::main]
-async fn main(_spawner: Spawner) {
-    let p = embassy_rp::init(Default::default());
+async fn main(spawner: Spawner) {
+    unsafe {
+        fat12_partition::init(&mut STORAGE, BLOCK_SIZE, BLOCKS);
+    }
```

Comment:
Perhaps we isolate the `unsafe` to show just the bit we're unsafeing:
```suggestion
    fat12_partition::init(unsafe { &mut STORAGE }, BLOCK_SIZE, BLOCKS);
```

(and at the end of this file too)